pilot/pkg/networking/core/accesslog.go

```diff
@@ -323,7 +323,7 @@ func (b *cachedMeshConfigAccessLog) buildOrFetch(mesh *meshconfig.MeshConfig) *a
 		return c
 	}
 	// We need to build access log. This is needed either on first access or when mesh config changes.
-	accessLog := model.FileAccessLogFromMeshConfig(mesh.AccessLogFile, mesh)
+	accessLog := model.FileAccessLogFromMeshConfig(mesh.AccessLogFile, mesh, true)
```

Comment:
Shouldn't this be proxy dependent?

Reply:
Here we may hit the cache, so it's meaningless to check version, we assume that it's 1.26+(which should be the most of cases).

Reply:
Can we make non-1.26+ proxies bypass/break the cache? I'm just worried about the failure mode here

Reply:
Or we can cached by proxy version, WDYT?

Reply:
Cache per proxy version makes sense; thanks!

Reply:
rethink about it, it's overkill to cache per version, use bool as key.